core/src/main/kotlin/defaultConfiguration.kt

```diff
@@ -37,19 +40,21 @@ data class DokkaSourceSetImpl(
     override val apiVersion: String? = null,
     override val noStdlibLink: Boolean = DokkaDefaults.noStdlibLink,
     override val noJdkLink: Boolean = DokkaDefaults.noJdkLink,
-    override val suppressedFiles: List<String> = emptyList(),
+    override val suppressedFiles: List<File> = emptyList(),
     override val analysisPlatform: Platform = DokkaDefaults.analysisPlatform
 ) : DokkaSourceSet
 
 data class DokkaModuleDescriptionImpl(
     override val name: String,
-    override val path: String,
-    override val docFile: String
+    override val path: File,
+    override val docFile: File
 ) : DokkaConfiguration.DokkaModuleDescription
 
 data class SourceRootImpl(
-    override val path: String
-) : DokkaConfiguration.SourceRoot
+    override val directory: File
+) : DokkaConfiguration.SourceRoot {
+    constructor(directoryPath: String) : this(File(directoryPath))
+}
```

Comment:
IMHO we can drop the `SourceRoot` abstraction entirely. It's there for historical reasons (it had more than one property) but it makes Gradle configuration harder (you can't just pass a list of files to `sourceRoots` property) and I personally don't see any advantages of having this wrapper

Reply:
👍 Very, very reasonable!

Reply:
Done ✅